ui/src/app/shared/header/header.component.html

```diff
@@ -10,14 +10,26 @@
              src="https://www.gstatic.com/images/icons/material/system/1x/cancel_grey600_18dp.png">
       </mat-chip>
       <mat-menu #menu="matMenu" [overlapTrigger]="false" class="wide-menu">
-        <mat-form-field class="query-input">
+        <mat-form-field *ngIf="isFreeText()" class="query-input">
```

Comment:
opt: `*ngSwitch` might be nice here and would ensure mutual exclusion for these being displayed